bpo-45584: Clarify the meaning of `truncate` in documentation (#29183)
python/cpython: Doc/library/math.rst

.. function:: trunc(x)

>    Return *x* with the fractional part removed, leaving the integer part.  This
>    rounds toward 0 and is equivalent to floor and ceil for positive and negative
>    *x* respectively.  If *x* is not a float, delegates to :meth:`x.__trunc__()
>    <object.__trunc__>`, which should return an :class:`~numbers.Integral` value.

Reviewer: ```suggestion
   Given a :class:`~numbers.Real` value *x*, return *x* with the fractional part
   removed, leaving the integer part.  This rounds toward 0: ``trunc()`` is
   equivalent to :func:`floor` for positive *x*, and equivalent to :func:`ceil` for
   negative *x*. If *x* is not a float, ``trunc()`` delegates to
   :meth:`x.__trunc__()<object.__trunc__>`, which should return an
   :class:`~numbers.Integral` value if properly defined.
```

.. function:: trunc(x)

>    Given a :class:`~numbers.Real` value *x*, return *x* with the fractional part

Reviewer 2: The following is what I wrote on the issue.
```suggestion
   Return *x* with the fractional part
```

Reviewer 3: Agree, this is consistent with floor() and ceil() and also more accurate because this function doesn't directly interact with the `numbers` module.

   removed, leaving the integer part.  This rounds toward 0: ``trunc()`` is
   equivalent to :func:`floor` for positive *x*, and equivalent to :func:`ceil` for
   negative *x*. If *x* is not a float, ``trunc()`` delegates to

>    :meth:`x.__trunc__ <object.__trunc__>`, which should return an

Reviewer 4: I have suggested restoring the space here and committed the suggestion.

Also removed the parens, which are added to all function references by the sphinx config (wrongly in my opinionn, but at least it’s configurable in one place).

Reviewer 2: In my test, :func: adds '()', :method: does not.  I like it better without.  See my separate review and suggestion about being consistent with linking.

Reviewer 2: Devguide has `:ref:`link text <reference-label>`` with space.

>    :meth:`x.__trunc__()<object.__trunc__>`, which should return an

Reviewer 4: ```suggestion
   :meth:`x.__trunc__ <object.__trunc__>`, which should return an
```

>    negative *x*. If *x* is not a float, ``trunc()`` delegates to
>    :meth:`x.__trunc__ <object.__trunc__>`, which should return an
>    :class:`~numbers.Integral` value if properly defined.

Reviewer 2: It was my intention that this sentence be a copy of the corresponding sentences in the ceil and floor entries.
 The latter reads "If *x* is not a float, delegates to ``x.__floor__()``, which should return an
:class:`~numbers.Integral` value."

```suggestion
   negative *x*. If *x* is not a float, delegates to
   :meth:`x.__trunc__ <object.__trunc__>`, which should return an
   :class:`~numbers.Integral` value.
```
I left the markup change, though, since it makes the method name link to the Data model entry.  I think that this is useful, but if we do it here, we should do it for ceil and floor.  Otherwise, make a separate issue/PR for making the code a link.

.. function:: trunc(x)

   Return the :class:`~numbers.Real` value *x* truncated to an
   :class:`~numbers.Integral` (usually an integer). Truncating *x* means

>    removing the digits after the decimal comma, hence rounding toward 0. It is

Reviewer 5: Since not all locales use a comma:
```suggestion
   removing the digits after the decimal separator, hence rounding toward 0. It is
```

Author: Done

Author: Done. 
I didn't realize that the English doc had to be normalized. I admit that if I were to translate it to French (my mother tongue), I clearly would have used "dot". But I guess it's great this way, thanks!

> Not a math expert but the change makes sense to me!

Great! That's the point. That the doc is easy to get even without being a math expert. At least in my opinion.
